# Generated by Django 5.2.1 on 2025-05-14 17:32

import django.db.models.functions.text
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('classification', '0002_filetype_unique_lower_extension_software'),
        ('organization', '0002_alter_organization_abbr_alter_organization_ror_id_and_more'),
        ('research', '0002_alter_resource_curators_and_more'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.AlterModelOptions(
            name='resource',
            options={'verbose_name': 'Research resource', 'verbose_name_plural': 'Research resources'},
        ),
        migrations.AddConstraint(
            model_name='resource',
            constraint=models.UniqueConstraint(django.db.models.functions.text.Lower('slug'), name='unique_lower_slug', violation_error_message='A research resource with this slug already exists. This validation is case-insensitive.'),
        ),
        migrations.AddConstraint(
            model_name='resource',
            constraint=models.UniqueConstraint(django.db.models.functions.text.Lower('title_en'), condition=models.Q(('title_en', ''), _negated=True), name='unique_lower_title_en', violation_error_message='A research resource with this title (EN) already exists. This validation is case-insensitive.'),
        ),
        migrations.AddConstraint(
            model_name='resource',
            constraint=models.UniqueConstraint(django.db.models.functions.text.Lower('title_de'), condition=models.Q(('title_de', ''), _negated=True), name='unique_lower_title_de', violation_error_message='A research resource with this title (DE) already exists. This validation is case-insensitive.'),
        ),
    ]
